flocker/node/script.py

```diff
@@ -113,32 +109,42 @@ def agent_config_from_file(path):
                     },
                     "port": {"type": "integer"},
                 }
+            },
+            "dataset": {
+                "type": "object",
+                "oneOf": [
+                    {
+                        "required": ["backend"],
+                        "properties": {
+                            "backend": {
+                                "type": "string",
+                                "pattern": "zfs",
+                            },
+                            "zfs-pool": {
```

Comment:
They can't - it is "oneOf" - that reminds me that there is a missing test. I did it like this to match the `zfs-pool: flocker` in the accepted design in https://github.com/ClusterHQ/flocker/pull/1338.